examples/tensorize_vllm_model.py

```diff
@@ -0,0 +1,275 @@
+import argparse
+import os
+import tempfile
+import time
+import uuid
+from functools import partial
+from typing import Type
+
+import torch
+import torch.nn as nn
+from tensorizer import (DecryptionParams, EncryptionParams, TensorDeserializer,
+                        TensorSerializer, stream_io)
+from tensorizer.utils import convert_bytes, get_mem_usage, no_init_or_tensor
+from transformers import AutoConfig, AutoModelForCausalLM, PretrainedConfig
+
+from vllm.config import _get_and_verify_dtype
+from vllm.distributed import initialize_model_parallel
+from vllm.model_executor.models import ModelRegistry
+from vllm.model_executor.tensorizer_loader import TensorizerArgs
+
+# yapf conflicts with isort for this docstring
+# yapf: disable
+"""
+tensorize_vllm_model.py is a script that can be used to serialize and 
+deserialize vLLM models. These models can be loaded using tensorizer directly 
+to the GPU extremely quickly. Tensor encryption and decryption is also 
```

Comment:
Just a nit: This is slightly misleading. The serialization step is done for `AutoModelForCausalLM` model object but we verify if it can be supported by `vllm`  only in deserialization step.

Reply:
The model serialized in this script is done for a model instantiated as a vLLM model.

```
def serialize():
    model = AutoModelForCausalLM.from_pretrained(model_ref)
    config = AutoConfig.from_pretrained(model_ref)
    make_model_contiguous(model)
    to_dtype = _get_and_verify_dtype(config, dtype=dtype)
    model = model.to(dtype=to_dtype)
    with tempfile.TemporaryDirectory() as tmpdir:
        DOWNLOAD_DIR = os.path.join(tmpdir, model_name)
        model.save_pretrained(DOWNLOAD_DIR)
        del model
        model_class = _get_vllm_model_architecture(config)
```

Where `_get_vllm_model_architecture` is defined as

```
def _get_vllm_model_architecture(config: PretrainedConfig) -> Type[nn.Module]:
    architectures = getattr(config, "architectures", [])
    for arch in architectures:
        model_cls = ModelRegistry.load_model_cls(arch)
        if model_cls is not None:
            return model_cls
    raise ValueError(
        f"Model architectures {architectures} are not supported for now. "
        f"Supported architectures: {ModelRegistry.get_supported_archs()}")
```

Which makes use of `ModelRegistry.load_model_cls` to instantiate a vLLM model for serialization, and loads the config into it.

Reply:
Ah I see! Sorry about not reading it carefully!

Reply:
No worries! I should have all of your comments resolved with a commit coming briefly. It also uses `LLMEngine` here directly to ensure all of vLLM's model loading optimizations and architecture are used.